### ISSUE TITLE
Values in the YAML file are missing single quotes

### ISSUE THREAD

**Issue by renards** [clivebeckett/grav-plugin-admin-addon-media-metadata#4]
Hello,

first of all thanks for this plugin, which makes my life a bit easier. :grinning: 

I tried to use a string like 'Photo: test' for an alternative text, but the YAML linter does not like colons in a value.
So I modified your code a bit to use single quotes for the values. If you are interested, I can submit a PR but I think it's better to checkout a develop branch first to merge modifications into it.

Cheers

**Comment by clivebeckett** [comment]
Hi @renards!

thanks a lot! I have added a develop branch and I am happy to receive PRs. I have only very little experience with that but I guess the github interface will help me :)

As far as I see in your fork you have implemented a way to replace the YAML parsing with Grav core technology? That would be amazing!

Cheers,
Clive.

**Comment by clivebeckett** [comment]
Hi again,

I’m sorry for asking such probably stupid questions, still being new to collaborative development. What if I want to test the code from your Pull Request in my local version before I approve it and merge it into my develop branch? How do I do that? These ones are relatively small and I’m sure you tested everything :) But I guess in general it makes sense to test someone elses PR myself.

**Comment by renards** [comment]
Hi Clive,

no, that's not a stupid question and yes you are right : it's important to test the code before merging. I don't have that much experience with GitHub either. I worked more with GitLab. 

In the GitHub docs you can find this article : https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/checking-out-pull-requests-locally .
So in this case you can try the following :
`git fetch origin pull/5/head:task/modify-writing-of-the-yaml-file`
`git checkout task/modify-writing-of-the-yaml-file`

Then you should be able to test the changes locally.
Hope this helps.

**Comment by clivebeckett** [comment]
@renards Thanks a lot for all your help, not only by your contribution but by helping me with reviewing your pull request.

I have changed two more things:

- I added the way that you implemented for editing the *.meta.yaml (editMetaDataFile()) file, into the method createYamlFile() as well
- I added a small fix to the parseYamlFile(): due to the single quotes that were implemented by your commits, a YAML data value that was not edited in the form got single quotes added every time the form was submitted

All the changes are merged into both the develop and the master branch.

Cheers,
Clive.

**Comment by renards** [comment]
Hi Clive,

you are welcome !

I have seen your changes and they are almost the same as I have already made. :wink:
Only one thing : I would declare the var `$newMetaData` before using it and place it before the code block starting at line 234.

I will prepare another PR for the issue #1 in the next few days.

Cheers

**Comment by clivebeckett** [comment]
Hi again,

the way I implemented writing the YAML file in `createMetaYaml()` is _exactly_ the same as yours 😉 – I copied it from your changes in `editMetaDataFile()` … and I tried to say that in my last reply. And thanks for your note about declaring the array before using it. I often forget that.

Thanks to your implementation of `Grav\Common\Yaml` I now also found the way to parse the *.meta.yaml using this core class as well. My own `parseYamlFile()` is thus not needed anymore. I’ll commit and push it soon, together with your suggested bugfix of declaring `$newMetaData` before the foreach loop. I’ll need to take care of the CHANGELOG first.

I’m really grateful for your commitment. I’ve been experiencing remote collaboration in my music but this is the first time on a programming project … and I am enjoying it :)

Cheers,
Clive.
